### ISSUE TITLE
Make content `date` handling more generic

### ISSUE THREAD

**Issue by bmcminn** [reptar/reptar#39]
The default `_posts/my-first-yarn.md` file generated by `yarn init` contains a `date` property in the YAML frontmatter config with the date and time it was created (during `yarn init`).

```md
---
title: Example Page Title Here
date: 2016-1-5T15:01:24.105Z
---

Example content...
```

A few concerns regarding this undocumented behavior:

1. The issue I find is that Yarn expects the date string to be a fully qualified date string as seen above, which isn't realistic for average users to define without using the `yarn new file` scaffolding tool.
1. Invalid `date` string values (ie: `date: 2016-5-30`) causes Yarn to default to the current system date.
1. The scaffolding tool mentioned earlier prepends the current system date the file was created to the filename, which overrides the `date` property and prevents the user from intuitively updating the date property.
1. Setting the `date` property to a future date (eg: `date: 2020-5-20T00:00:00.105Z`) causes Yarn to silently fail in generating `index.html` and post content files in `_site/` after running `yarn clean`. _I would expect Yarn to ignore generating this file until that date has lapsed._
    - Changing the `date` property on a file to a future date does not remove the old generate file from the file system. _If this is the intended behavior for Yarn, I would expect it to be documented as such._
    - This will likely need to be a separate issue in conjunction with this one.

If the above issues should be broken out into their own separate issues, let me know and I can file them accordingly.

**Comment by hswolff** [comment]
Thanks for opening this issue @bmcminn!

Definitely agree that the handling for `date` fields should be more lenient. I'd love to add that into yarn itself so that `date: 2016-5-30` works as expected. 

> The scaffolding tool mentioned earlier prepends the current system date the file was created to the filename, which overrides the date property and prevents the user from intuitively updating the date property.

That's a hard question that I struggled with when implementing that logic - what should take precedence? Should it be the file system file-name or the frontmatter in the file? Should it be configurable? I don't know!

> Setting the date property to a future date (eg: date: 2020-5-20T00:00:00.105Z) causes Yarn to silently fail in generating index.html and post content files in _site/. I would expect Yarn to ignore generating this file until that date has lapsed.

That's surprising as that should work. That definitely sounds like a bug. I'm curious if the issue is: is it not properly filtering out future dates, or if there are no posts then the build process fails?

**Comment by bmcminn** [comment]
I agree the filename date should take precedence. That's a more explicit action to rename a file as opposed to editing its frontmatter, though I think this should be documented.

As for the future date issue on Windows, I had three different post files, each with a different `date` config, one of them was a future date. After I cleaned the public directory and ran `yarn build` the `index.html` file and posts failed to render. Possibly a bug, though I'm not sure where in the stack this would take place.

**Comment by hswolff** [comment]
> I agree the filename date should take precedence. That's a more explicit action to rename a file as opposed to editing its frontmatter, though I think this should be documented.

So I just checked the code and the filename actually has no influence on where the date is taken from. Whoops! It's only taken from the frontmatter. So that's one issue we don't have to worry about! 😌 

> As for the future date issue on Windows, I had three different post files, each with a different date config, one of them was a future date. After I cleaned the public directory and ran yarn build the index.html file and posts failed to render. Possibly a bug, though I'm not sure where in the stack this would take place.

Could you maybe zip up the posts and paste them here for me to try locally? I tried making one post in the future and no `index.html` file is rendered as the Collection is empty. This is expected behavior, however maybe not desired. It sounds like an error is being thrown for you though?

**Comment by hswolff** [comment]
Current action items I'm taking for this issue are:

- [ ] Make date handling more generic. In other words support `2016-10-10` instead of the full ISO time structure.

**Comment by hswolff** [comment]
Ok! I got this fixed in the soon to be released `v2.0.0` release!

Read full details in this commit https://github.com/reptar/reptar/commit/7215ada9a767788a9967be75c88061e3de0cf239

This should make things simpler to reason about and behave as you expect now.
